platform/packages/platform/src/reply.rs

```diff
@@ -3,59 +3,83 @@
 }
 
-pub fn from_instantiate<T>(api: &dyn Api, reply: Reply) -> StdResult<InstantiateResponse<T>>
-where
-    T: DeserializeOwned,
-{
-    #[derive(Message)]
-    struct ReplyData {
-        #[prost(bytes, tag = "1")]
-        pub address: Vec<u8>,
-        #[prost(bytes, tag = "2")]
-        pub data: Vec<u8>,
+impl InstantiateResponse<Vec<u8>> {
+    fn into_addr(self) -> Addr {
+        self.address
     }
+}
 
-    let response = decode::<ReplyData>(reply)?;
+pub fn from_instantiate_addr_only(api: &dyn Api, reply: Reply) -> StdResult<Addr> {
+    from_instantiate_inner::<MsgInstantiateContractResponse>(api, reply)
+        .map(InstantiateResponse::into_addr)
+}
 
-    Ok(InstantiateResponse {
-        address: api
-            .addr_validate(&String::from_utf8(response.address).map_err(|_| {
-                StdError::generic_err("Address field contains invalid UTF-8 data!")
-            })?)?,
-        data: maybe_from_json(response.data)?,
-    })
+pub fn from_instantiate2_addr_only(api: &dyn Api, reply: Reply) -> StdResult<Addr> {
+    from_instantiate_inner::<MsgInstantiateContract2Response>(api, reply)
+        .map(InstantiateResponse::into_addr)
 }
 
 pub fn from_execute<T>(reply: Reply) -> StdResult<Option<T>>
 where
     T: DeserializeOwned,
 {
-    #[derive(prost::Message)]
-    struct ReplyData {
-        #[prost(bytes, tag = "1")]
-        pub data: Vec<u8>,
+    decode::<MsgExecuteContractResponse>(reply)
+        .map(|data| data.data)
+        .map(Binary)
+        .and_then(from_json)
+}
+
+struct UncheckedInstantiateResponse {
+    address: String,
+    data: Vec<u8>,
+}
+
+impl From<MsgInstantiateContractResponse> for UncheckedInstantiateResponse {
+    fn from(
+        MsgInstantiateContractResponse { address, data }: MsgInstantiateContractResponse,
+    ) -> Self {
+        Self { address, data }
     }
+}
 
-    decode::<ReplyData>(reply)
-        .map(|data| data.data)
-        .and_then(maybe_from_json)
+impl From<MsgInstantiateContract2Response> for UncheckedInstantiateResponse {
+    fn from(
+        MsgInstantiateContract2Response { address, data }: MsgInstantiateContract2Response,
+    ) -> Self {
+        Self { address, data }
+    }
+}
+
+fn from_instantiate_inner<R>(api: &dyn Api, reply: Reply) -> StdResult<InstantiateResponse<Vec<u8>>>
+where
+    R: Message + Default + Into<UncheckedInstantiateResponse>,
+{
+    let UncheckedInstantiateResponse { address, data } = decode::<R>(reply)?.into();
+
+    api.addr_validate(&address)
+        .map(|address: Addr| InstantiateResponse { address, data })
 }
 
 fn decode_raw<M>(message: &[u8]) -> StdResult<M>
 where
     M: Message + Default,
 {
-    M::decode(message).map_err(|_| {
-        StdError::generic_err("Data is malformed or doesn't comply with used protobuf format!")
+    M::decode(message).map_err(|error| {
+        StdError::generic_err(format!("[Platform] Data is malformed or doesn't comply with used protobuf format! Cause: [Protobuf] {error}"))
```

Comment:
Platform != cosmwasm

Reply:
While true, that is an error that DOES within `platform`.